soh/src/code/z_player_lib.c

```diff
@@ -1325,38 +1325,50 @@ void func_80090A28(Player* this, Vec3f* vecs) {
     Matrix_MultVec3f(&D_80126098, &vecs[2]);
 }
 
-void Player_DrawHookshotReticle(PlayState* play, Player* this, f32 arg2) {
+void Player_DrawHookshotReticle(PlayState* play, Player* this, f32 hookshotRange) {
     static Vec3f D_801260C8 = { -500.0f, -100.0f, 0.0f };
-    CollisionPoly* sp9C;
+    CollisionPoly* colPoly;
     s32 bgId;
-    Vec3f sp8C;
-    Vec3f sp80;
-    Vec3f sp74;
+    Vec3f hookshotStart;
+    Vec3f hookshotEnd;
+    Vec3f firstHit;
     Vec3f sp68;
     f32 sp64;
-    f32 sp60;
 
     D_801260C8.z = 0.0f;
-    Matrix_MultVec3f(&D_801260C8, &sp8C);
-    D_801260C8.z = arg2;
-    Matrix_MultVec3f(&D_801260C8, &sp80);
+    Matrix_MultVec3f(&D_801260C8, &hookshotStart);
+    D_801260C8.z = hookshotRange;
+    Matrix_MultVec3f(&D_801260C8, &hookshotEnd);
 
-    if (BgCheck_AnyLineTest3(&play->colCtx, &sp8C, &sp80, &sp74, &sp9C, 1, 1, 1, 1, &bgId)) {
+    if (BgCheck_AnyLineTest3(&play->colCtx, &hookshotStart, &hookshotEnd, &firstHit, &colPoly, 1, 1, 1, 1, &bgId)) {
         OPEN_DISPS(play->state.gfxCtx);
 
         WORLD_OVERLAY_DISP = Gfx_SetupDL(WORLD_OVERLAY_DISP, 0x07);
 
-        SkinMatrix_Vec3fMtxFMultXYZW(&play->viewProjectionMtxF, &sp74, &sp68, &sp64);
+        SkinMatrix_Vec3fMtxFMultXYZW(&play->viewProjectionMtxF, &firstHit, &sp68, &sp64);
 
-        sp60 = (sp64 < 200.0f) ? 0.08f : (sp64 / 200.0f) * 0.08f;
+        const f32 sp60 = (sp64 < 200.0f) ? 0.08f : (sp64 / 200.0f) * 0.08f;
 
-        Matrix_Translate(sp74.x, sp74.y, sp74.z, MTXMODE_NEW);
+        Matrix_Translate(firstHit.x, firstHit.y, firstHit.z, MTXMODE_NEW);
         Matrix_Scale(sp60, sp60, sp60, MTXMODE_APPLY);
 
-        gSPMatrix(WORLD_OVERLAY_DISP++, MATRIX_NEWMTX(play->state.gfxCtx),
-                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
-        gSPSegment(WORLD_OVERLAY_DISP++, 0x06, play->objectCtx.status[this->actor.objBankIndex].segment);
-        gSPDisplayList(WORLD_OVERLAY_DISP++, gLinkAdultHookshotReticleDL);
+        gSPMatrix(WORLD_OVERLAY_DISP++, MATRIX_NEWMTX(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        char* tex = ResourceMgr_LoadTexOrDListByName(gLinkAdultHookshotReticleTex);
```

Comment:
Two things, 

A) It's not immediately clear why this couldn't have been done with the dlist patching method we use for other cosmetics instead of bringing our own texture

B) We're not preserving authentic behavior here, we should only diverge here if the user has changed the cosmetics color or opted in to the "Targetable Hookshot Reticle" option

Reply:
For your first concern, I didn't bring in a texture. It references the original texture that was in the static DL.  For the second, If the option is disabled it will still act the same, doing it the way I am allows setting a custom color without needing to enable targetable reticle. 
IMO while its not the same code it is still functionally equivalent with default settings which is fine for me. I also like having the consistency with how colors are done for gauntlets and tunics.

Reply:
Ah I see now, so we're effectively inline-ing the contents of `gLinkAdultHookshotReticleDL` 1:1 aside from the color change?

Reply:
Yes. I may have pulled out a useless command or two but its basically the same.